modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_builder.py

```diff
@@ -906,7 +906,10 @@ def _process_groupby(self, op):
             bool_cols := {
                 c: cast_agg[agg_exprs[c].agg]
                 for c, t in frame.dtypes.items()
-                if is_bool_dtype(t) and agg_exprs[c].agg in cast_agg
+                # Do not call is_bool_dtype() for categorical since it checks all the categories
```

Comment:
I'm wondering whether we can hit this in other places? @dchigarev what do you think?

Reply:
> Do not call is_bool_dtype() for categorical since it checks all the categories

Does this problem (slow `is_bool()` call) occur only with our [lazy categories](https://github.com/modin-project/modin/blob/3b3bba9a83b6bec35e86826800004987afe24d47/modin/core/dataframe/pandas/metadata/dtypes.py#L155) or with native pandas categories as well?

Reply:
With native pandas categories it's also slower, because it calls `dtype.categories.inferred_type` and it's much slower with lazy categories, because it causes the categories materialization.